src/app/data-planes/data/index.ts

```diff
@@ -277,9 +288,8 @@ export const Rule = {
         // with a single entry so it looks like to and from rules
         const proxy = typeof item.proxyRule !== 'undefined'
           ? [{
-            ...Rule.fromObject(item.proxyRule as InspectBaseRule),
+            ...Rule.fromObject(item.proxyRule as InspectBaseRule, 'proxy'),
```

Comment:
I would rather these methods always took a single argument, this is why previously the ruleType was applied here instead of inside `Rule`. Moreover if we are still adding the `type` here also we may aswell continue to add the `ruleType` here.

If it gets to a point where we absolutely must pass extra things in then fair enough, but I don't think we at that point.

Reply:
There was a specific reason why I needed to move the addition of `ruleType` inside the transformer: I wanted a stricter type for `ruleType` so that autocompletion of the field knows of the three possible values. That meant that in order for `fromObject` to return a `Rule`, it needs to set one of the three allowed values (`''` not being one of them). And that can only happen inside the `fromObject` function.

Now, theoretically `type` could also be pulled into the function, but for `type` there is neither an advantage nor a forcing factor: it doesn’t come from `rule` and its resulting type will only be `string` anyway. While it would be nice to have a `PolicyTypeName` union type of all policy type names to make this more precise, it would require quite a bit more work to establish one.

Reply:
`type` should always be string seeing as policies are pluggable.

So your reasoning here not to have `'' | 'to' | 'from' | 'proxy'` is incase you accidently search for a `ruleType=''`? (which would return none)

Reply:
> So your reasoning here not to have `'' | 'to' | 'from' | 'proxy'` is incase you accidently search for a `ruleType=''`? (which would return none)

Why would I allow `''` in the first place? It’s not useful and I can easily avoid it by pulling the rule type stuff inside.

Reply:
Well yeah but if ideally we want to stick to the single argument `fromObject` interface, it kinda depends what we prefer to keep. I'm not sure in what circumstance anyone would try and search ruleType by `''` even if Typescript allowed them to do that.

We could also have a private type in the middle to allow us to create a rule without a `ruleType` and then add the type where we did previously. Let me look/think about this a little more.

Btw this isn't a blocker.

Reply:
What if we used `to` as a default instead of `` inside `Rule.FromObject`, then (as was the case previous to the changes here) we immediately overwrite it with the value from the API always. That way we get the types you are after and we keep the single argument `fromObject` API?

There are other ways to do this (we could have multiple `To/From/ProxyRule.fromObject` etc) but I think this is the simplest and I think it answers both concerns, wdyt?

Reply:
I think that’d be acceptable. I’ve changed this now.